sdk/compiler/daml-lf-tools/src/DA/Daml/LF/TypeChecker/Upgrade.hs

```diff
@@ -870,11 +871,10 @@ checkQualType deps name =
                   || case (mbPastVersion, mbPresentVersion) of
                       (Just pastVersion, Just presentVersion) -> pastVersion < presentVersion
                       (Nothing, Nothing) -> True -- Only daml-prim has no version field
-                      _ -> True -- This case should not be possible
-                _ -> True -- This case should not be possible
-            _ -> False -- Cannot go from 1.15 (non-upgradeable) to 1.17 (upgradeable)
-        _ ->
-          False
+                      _ -> False -- Not possible
+                _ -> False -- Not possible to have types from utility packages
+            _ -> False -- Shouldn't go from 1.15 (non-upgradeable) to 1.17 (upgradeable)
+        _ -> False -- Not possible
```

Comment:
Might be worth saying why these aren't possible, since you have 5 cases now that aren't explained

Reply:
I'll defer this to the next PR, when each impossible case will have an explanation regardless